lib/ecto/migration/schema_migration.ex

```diff
@@ -38,4 +28,42 @@ defmodule Ecto.Migration.SchemaMigration do
   def down(repo, version) do
     repo.delete_all from(p in __MODULE__, where: p.version == ^version), @opts
   end
+
+  defp create_migrations_table(Ecto.Adapters.MySQL = adapter, repo) do
```

Comment:
If I use `create_if_not_exists` here, I get errors that have to do with savepoints in MYSQL. Using the old approach for MySQL fixes it. This was the weird error that you and @liveforeverx were discussing in irc. `create_if_not_exists` for MySQL works fine outside of schema migrations though.

Should we revert `ddl_exists?` so we can use it here? Any idea on how we can use `create_if_not_exists` for MySQL here?